ciborium/tests/codec.rs

```diff
@@ -307,19 +307,53 @@ fn codec<'de, T: Serialize + Clone, V: Debug + Eq + Deserialize<'de>, F: Fn(T) -
 
     let decoded: Value = from_reader(&bytes[..]).unwrap();
     eprintln!("{:x?} == {:x?}", &value, &decoded);
-    assert_eq!(value, decoded);
+    assert!(veq(&value, &decoded));
 
     let decoded: V = value.deserialized().unwrap();
     let answer = equality(input);
     eprintln!("{:x?} == {:x?}", answer, decoded);
     assert_eq!(answer, decoded);
 }
 
+#[inline]
+fn veq(lhs: &Value, rhs: &Value) -> bool {
+    if let Value::Float(l) = lhs {
+        if let Value::Float(r) = rhs {
+            return Float(*l) == Float(*r);
+        }
+    }
+
+    lhs == rhs
+}
+
 #[inline]
 fn same<T>(x: T) -> T {
     x
 }
 
+#[derive(Debug, Deserialize)]
+struct Float<T>(T);
+
+impl PartialEq for Float<f32> {
+    fn eq(&self, other: &Float<f32>) -> bool {
+        if self.0.is_nan() && other.0.is_nan() {
+            return true;
+        }
+
+        self.0 == other.0
+    }
+}
+
+impl PartialEq for Float<f64> {
+    fn eq(&self, other: &Float<f64>) -> bool {
+        if self.0.is_nan() && other.0.is_nan() {
+            return true;
+        }
+
+        self.0 == other.0
+    }
+}
```

Comment:
Hmm, reading the doc of `PartialEq`:

> For example, in floating point numbers NaN != NaN, so floating point types implement PartialEq but not Eq.

shouldn't this also implement `ne()` with:

```rust
    fn ne(&self, other: &Float<f32>) -> bool {
        if self.0.is_nan() && other.0.is_nan() {
            return true;
        }

        self.0 != other.0
    }
```